Classes/Command/ValidatorCommand.php

```diff
@@ -0,0 +1,168 @@
+
+    private function askForValidatorInformation(SymfonyStyle $io, InputInterface $input): ValidatorInformation
+    {
+        $extensionInformation = $this->getExtensionInformation(
+            $this->askForExtensionKey($io, $input->getArgument('extension_key')),
+            $io
+        );
+
+        $name = $this->askForValidatorName($io);
+        $type = $this->askForValidatorType($io);
+        $model = null;
+        if ($type === ValidatorType::MODEL) {
+            $model = $this->askForValidatorModel($io, $extensionInformation);
+        }
+
+        return new ValidatorInformation(
+            $extensionInformation,
+            $name,
+            $type,
+            $model
+        );
+    }
+
+    private function askForValidatorType(SymfonyStyle $io): ValidatorType
+    {
+        $choices = array_map(
+            fn(ValidatorType $type) => $type->value,
+            ValidatorType::cases()
+        );
+
+        $selected = $io->choice(
+            'Please select the type of your Validator',
+            $choices
+        );
+
+        // Convert selected string back to enum
+        return ValidatorType::from($selected);
+    }
+
+    private function askForValidatorName(SymfonyStyle $io): string
+    {
+        $defaultName = null;
+        do {
+            $name = (string)$io->ask(
+                'Please provide the name of your Validator',
+                $defaultName,
+            );
+
+            if (preg_match('/^\d/', $name)) {
+                $io->error('Validator name should not start with a number.');
+                $defaultName = $this->tryToCorrectValidatorName($name);
+                $validValidatorName = false;
+            } elseif (preg_match('/[^a-zA-Z0-9]/', $name)) {
+                $io->error('Validator name contains invalid chars. Please provide just letters and numbers.');
+                $defaultName = $this->tryToCorrectValidatorName($name);
+                $validValidatorName = false;
+            } elseif (preg_match('/^[a-z0-9]+$/', $name)) {
+                $io->error('Validator must be written in UpperCamelCase like BlogExampleValidator.');
+                $defaultName = $this->tryToCorrectValidatorName($name);
+                $validValidatorName = false;
+            } elseif (!str_ends_with($name, 'Validator')) {
+                $io->error('Validator must end with "Validator".');
+                $defaultName = $this->tryToCorrectValidatorName($name);
+                $validValidatorName = false;
+            } else {
+                $validValidatorName = true;
+            }
+        } while (!$validValidatorName);
+
+        return $name;
+    }
+
+    private function tryToCorrectValidatorName(string $givenName): string
+    {
+        // Remove invalid chars
+        $cleanedName = preg_replace('/[^a-zA-Z0-9]/', '', $givenName);
+
+        // Upper case first char
+        $cleanedName = ucfirst($cleanedName);
+
+        // Remove ending "cOntroLLeR" with wrong case
```

Comment:
Be creative ;-) `VaLIdaTOr`?

Reply:
as if that matters. We should extract this method anyway